### PR TITLE
New feature: support WalletConnect

### PR DESCRIPTION
**Pull request by Sven-TBD**
- [ ] Project Research & Management #144 
    - [ ] #148 
- [x] [PRD](https://vs0cjf.axshare.com/#g=1&id=psrijc&p=walletconnect-general)
- [ ] [UI](https://www.figma.com/file/6XNoimRDbFTTNm016rbIdU/Magickbase?type=design&node-id=16072%3A38648&t=fpEwJ6TlIk4l2WFi-1)
- [ ] Neuron GUI 
    - [x] #153 
- [ ] Backend
- [ ] Discussion
    - [x] #220 
    - [x] #197  
    - [ ]  #212 
    - [x] #213
    - [x] #215
    - [x] #214

### CODE REVIEW

**Comment by Sven-TBD** [comment]
@Kirl70  I have updated the Send Tx section of the Axure link, could you pls design it before next monday?
image

**Comment by Kirl70** [comment]
> @Kirl70 I have updated the Send Tx section of the [Axure link](https://vs0cjf.axshare.com/#g=1&id=psrijc&p=walletconnect-general), could you pls design it before next monday? image

Design Draft ( 3-1 ~ 3-6 ): 
https://www.figma.com/file/6XNoimRDbFTTNm016rbIdU/Magickbase?type=design&node-id=16536%3A38593&t=QsMaQAtJrw34nlCb-1
image

File: Neuron-PRDs/WalletConnect/GeneralComponent.md [hunk 1]
```diff
@@ -44,3 +44,30 @@ To cancel this request, just press the Reject button.
...
+- Click the pencil like icon
+![picture 3](pic/dde9d7d213ce52ce6b1f9febcfc2585cc743ff7ac0d0938cbcd64de595092660.png)  
+
+- Then modify information in corresponding text field, and click NEXT to continue.
+
+#### Further examine the receiving address and its detailed data
+
+- Click the arrow next to the address. The Data and Witness will display(elaborately speaking, these details will move upwards)
+![picture 4](pic/70d8140d96601991d8aa2632fc65d0c2a1eccef17ed02505fe50b9adade86115.png)  
+- In the UI demo, click this arrow, users can view all the receiving addresses, then view the data and witness the one you want by click the arrow next to the address on list.
```
Comment (Keith-CY) [inline Neuron-PRDs/WalletConnect/GeneralComponent.md:72]: The detail should be described by text clearly and abstractly.
`click this arrow` is concrete, `check more addresses` is abstract

Comment (Keith-CY) [reply Neuron-PRDs/WalletConnect/GeneralComponent.md:72]: How is the abbreviation of address list shown should be described.

In the screenshot, `To address: ckt...s49 / ckt...82 / ...` is meaning less.
What if show the addresses in various format based on the count:
1. only 1 address: display the full address
2. more than 1 address: show the count of addresses and check address list in another view.

File: Neuron-PRDs/WalletConnect/FAQ.md
Comment (Sven-TBD) [inline Neuron-PRDs/WalletConnect/FAQ.md:1]: Lock Hash = lock script 里的 code hash+lock script 里的 hash type+ lock script 里的 argus
其实这里只需要 lock script 里的 code hash，严格来讲，lock script 里的 hash type也是要的，只是冲突较少。

Comment (Sven-TBD) [reply Neuron-PRDs/WalletConnect/FAQ.md:1]: 上下文中提到的Auth 指的是 lock script 里的 code hash。

Comment (Sven-TBD) [reply Neuron-PRDs/WalletConnect/FAQ.md:1]: 本协议中的交互双方分别是：热钱包和dapp。离线钱包、观察钱包的交互不在本协议的讨论范围内。

**Pull request merged**